src/Guide/Api/Methods.hs

```diff
@@ -112,53 +123,66 @@ createItem db catId name' = do
       looksLikeLibrary = T.all isAllowedChar name'
       kind' = if looksLikeLibrary then Library (Just name') else Other
   time <- liftIO getCurrentTime
-  (_edit, _newItem) <- dbUpdate db (AddItem catId itemId name' time kind')
-  -- TODO: addEdit edit
+  (_edit, _) <- dbUpdate db (AddItem catId itemId name' time kind')
+  config <- ask
+  addEdit db config requestDetails _edit
   pure itemId
 
 -- TODO: move an item
 
 -- | Set item's info
-setItemInfo :: DB -> Uid Item -> CItemInfo -> Guider NoContent
-setItemInfo db itemId CItemInfo{..} = do
+setItemInfo :: DB -> RequestDetails -> Uid Item -> CItemInfo -> Guider NoContent
+setItemInfo db requestDetails itemId CItemInfo{..} = do
   _ <- getItemOrFail db itemId
   -- TODO diff and merge
-  _ <- dbUpdate db $ SetItemName itemId $ unH ciiName
-  _ <- dbUpdate db $ SetItemGroup itemId $ unH ciiGroup
-  _ <- dbUpdate db $ SetItemLink itemId $ unH ciiLink
-  _ <- dbUpdate db $ SetItemKind itemId $ unH ciiKind
+  (_editName, _) <- dbUpdate db $ SetItemName itemId $ unH ciiName
+  (_editGroup, _) <- dbUpdate db $ SetItemGroup itemId $ unH ciiGroup
+  (_editLink, _) <- dbUpdate db $ SetItemLink itemId $ unH ciiLink
+  (_editKind, _) <- dbUpdate db $ SetItemKind itemId $ unH ciiKind
```

Comment:
Can be generalized, but I don't know, should we
```haskell
let edits =
  [ SetItemLink itemId $ unH ciiLink
  , SetItemGroup itemId $ unH ciiGroup
  , SetItemLink itemId $ unH ciiLink
  , SetItemKind itemId $ unH ciiKind
  ]
edits' <- mapM (fmap fst . dbUpdate db) edits
mapM_ (addEdit db config requestDetails) edits'
```

Reply:
edits list is heterogeneous, therefore it needed to add `dbUpdate db $ `. And it become
```haskell
let edits =
        [ fmap fst $ dbUpdate db $ SetItemName itemId $ unH ciiName
        , fmap fst $ dbUpdate db $ SetItemGroup itemId $ unH ciiGroup
        , fmap fst $ dbUpdate db $ SetItemLink itemId $ unH ciiLink
        , fmap fst $ dbUpdate db $ SetItemKind itemId $ unH ciiKind
        ]
```
Not so elegant.

Reply:
> Because I don't know, why there is a list?

The list is to apply `addEdit` function.